packages/app/models/order/fetcher.ts

```diff
@@ -1,143 +1,16 @@
-        symbol: "GNO"
-      },
-      createdAt: "1685616055",
-      startTime: 1693581600,
-      endTime: 1698420000,
-      orderSlots: [
-        "1693581600",
-        "1694186400",
-        "1694791200",
-        "1695396000",
-        "1696000800",
-        "1696605600",
-        "1697210400",
-        "1697815200"
-      ],
-      cancelledAt: "1685620710",
-      interval: "168"
-    },
-    {
-      id: "0x9a4b6a475370bb4f21c13fff7f54514751f24477",
-      owner: "0x05a4ed2367bd2f0aa63cc14897850be7474bc722",
-      receiver: "0x05a4ed2367bd2f0aa63cc14897850be7474bc722",
-      amount: "1999000000000000000",
-      sellToken: {
-        address: "0xe91d153e0b41518a2ce8dd3d7944fa863463a97d",
-        decimals: 18,
-        name: "Wrapped XDAI",
-        symbol: "WXDAI"
-      },
-      buyToken: {
-        address: "0x6a023ccd1ff6f2045c3309768ead9e68f978f6e1",
-        decimals: 18,
-        name: "Wrapped Ether on xDai",
-        symbol: "WETH"
-      },
-      createdAt: "1683890840",
-      startTime: 1683891419,
-      endTime: 1684495531,
-      orderSlots: [
-        "1683891419",
-        "1683977819",
-        "1684064219",
-        "1684150619",
-        "1684237019",
-        "1684323419",
-        "1684409819"
-      ],
-      cancelledAt: null,
-      interval: "24"
-    },
-    {
-      id: "0xc47bc680c00bc1620b0de3a638c565691cb616e8",
-      owner: "0x05a4ed2367bd2f0aa63cc14897850be7474bc722",
-      receiver: "0x05a4ed2367bd2f0aa63cc14897850be7474bc722",
-      amount: "419790000000000000000",
-      sellToken: {
-        address: "0xe91d153e0b41518a2ce8dd3d7944fa863463a97d",
-        decimals: 18,
-        name: "Wrapped XDAI",
-        symbol: "WXDAI"
-      },
-      buyToken: {
-        address: "0x9c58bacc331c9aa871afd802db6379a98e80cedb",
-        decimals: 18,
-        name: "Gnosis Token on xDai",
-        symbol: "GNO"
-      },
-      createdAt: "1685621305",
-      startTime: 1685621892,
-      endTime: 1693581600,
-      orderSlots: [
-        "1685621892",
-        "1686226692",
-        "1686831492",
-        "1687436292",
-        "1688041092",
-        "1688645892",
-        "1689250692",
-        "1689855492",
-        "1690460292",
-        "1691065092",
-        "1691669892",
-        "1692274692",
-        "1692879492",
-        "1693484292"
-      ],
-      cancelledAt: null,
-      interval: "168"
-    }
-  ]
-};
+import { GraphQLClient } from "graphql-request";
+import { filterAllButCancelledOrders } from "@/models/order/filters";
+import { getUserOrders, getSubgraphEndpoint, ChainId } from "@stackly/sdk";
 
-export async function getOrders(address: string) {
-  return mockOrders.orders;
+export async function getOrders(chainId: ChainId, address: string) {
+  try {
+    const graphqlClient = new GraphQLClient(getSubgraphEndpoint(chainId));
+    const orders = await getUserOrders(graphqlClient, address);
+
+    if (!orders) throw new Error("Failed to fetch SDK data");
```

Comment:
`"Failed to fetch subgraph data"` is the correct error message.

Reply:
If we change the underlying mechanism of the `SDK` this would also work.

So I think this error message is okay, we don't need to be super specific in this case. 

Let me know if you have a strong opinion on this and I'll change it. Otherwise let's keep it.

Reply:
I think we can be more specific. There are 2 ways we can fetch data. With subgraphs or querying contracts. I think we should be specific in this case, it can help debugging in the future. Also, yesterday I learned that we don't need to throw an Error object. We can simply do this:

```suggestion
    if (!orders) throw "Failed to fetch SDK data";
```

Reply:
Nice.

Okay, I'll update it then.